dask_ml/wrappers.py

```diff
@@ -202,13 +210,27 @@ def transform(self, X):
         """
         self._check_method("transform")
         X = self._check_array(X)
+        meta = self.meta
 
         if isinstance(X, da.Array):
-            xx = np.zeros((1, X.shape[1]), dtype=X.dtype)
-            dt = _transform(xx, self._postfit_estimator).dtype
-            return X.map_blocks(_transform, estimator=self._postfit_estimator, dtype=dt)
+            if meta is None:
+                xx = np.zeros((1, X.shape[1]), dtype=X.dtype)
+                dt = _transform(xx, self._postfit_estimator).dtype
+                return X.map_blocks(
+                    _transform, estimator=self._postfit_estimator, dtype=dt
+                )
+            else:
+                return X.map_blocks(
+                    _transform, estimator=self._postfit_estimator, meta=meta
+                )
```

Comment:
Does this work? If I understand [the docs](https://docs.dask.org/en/stable/generated/dask.array.map_blocks.html?highlight=map_blocks) correctly, that should hopefully do the trick.

```suggestion
            if meta is None:
                xx = np.zeros((1, X.shape[1]), dtype=X.dtype)
                dtype = _transform(xx, self._postfit_estimator).dtype
                meta = type(dtype)((), dtype=dtype)
            return X.map_blocks(
                _transform, estimator=self._postfit_estimator, meta=meta
            )
```

Reply:
Gotcha, Let me test this and  change it to your suggestion

Reply:
Thanks, Had to change the a bit but below does the trick. 
```python
  if meta is None:
       xx = np.zeros((1, X.shape[1]), dtype=X.dtype)
       x_t = _transform(xx, self._postfit_estimator)
       meta = type(x_t)((), dtype=x_t.dtype)
```
Resolving now.